active/teachingtip/TeachingTipSpec.md

```diff
@@ -0,0 +1,591 @@
+                    <TextBlock>You can change your tip preferences in <Hyperlink NavigateUri="app:/item/SettingsPage">Settings</Hyperlink> if you change your mind.</TextBlock>
+                </StackPanel>
+        </muxc:TeachingTip>
+    </muxc:TeachingTip.Attach>
+</Button>
+```
+
+C#
+```C#
+public void OnFirstSaveButtonClick(object sender, RoutedEventArgs args)
+{
+   AutoSaveTip.IsOpen = true;
+}
+```
+
+<a href="url"><img src="TeachingTipContentSample.jpg" height="600"></a>
+
+### Add buttons
+
+A close button is automatically added to the top of a teaching tip. However, a close button with custom text that is placed at the button of a teaching tip can be used instead by setting the TeachingTipCloseButtonKind property. A custom action button can also be added by setting ActionButtonText, ActionButtonCommand and the ActionButtonCommandParameter properties. 
+
+XAML
+```XAML
+<Button Content="Tip Example" Click="OnFirstSaveButtonClick">
+    <muxc:TeachingTip.Attach> 
+        <muxc:TeachingTip x:Name="AutoSaveTip"
+            Title="Saving automatically"
+            Subtitle="We save your changes as you go - so you never have to."
+            ActionButtonText="Disable"
+            ActionButtonCommand="DisableAutoSave"
+            CloseButtonKind="Footer"
+            CloseButtonText="Got it!"
+            Margin="15">
+                <CheckBox Content="Don't show tips at start up" Checked="HandleChecked" Unchecked="HandleUnchecked" />
+                <p>You can change your tip preferences in <a href="app:/item/SettingsPage">Settings</a> if you change your mind.</p>
+        </muxc:TeachingTip>
+    </muxc:TeachingTip.Attach>
+</Button>
+```
+
+C#
+```C#
+public void OnFirstSaveButtonClick(object sender, RoutedEventArgs args)
+{
+    TeachingTipExample.IsOpen=true;
+}
+```
+
+<a href="url"><img src="TeachingTipButtonsSample.jpg" height="600"></a>
+
+### Hero content
+
+Edge to edge content can be added to a teaching tip by setting the HeroContent property. The location of hero content can be set to the top or bottom of a teaching tip by setting the HeroContentPlacement property.
+
+XAML
+```XAML
+<Button Content="Tip Example" Click="OnFirstSaveButtonClick">
+    <muxc:TeachingTip.Attach> 
+        <muxc:TeachingTip x:Name="AutoSaveTip"
+            Title="Saving automatically"
+            Subtitle="We save your changes as you go - so you never have to."
+            <muxc:TeachingTip.HeroContent>
+                <Image Source="Assets/Giraffe.png" />
+            </muxc:TeachingTip.HeroContent>
+            Margin="15">
+        </muxc:TeachingTip>
+    </muxc:TeachingTip.Attach>
+</Button>
+```
+
+C#
+```C#
+public void OnFirstSaveButtonClick(object sender, RoutedEventArgs args)
+{
+    TeachingTipExample.IsOpen=true;
+}
+```
+
+<a href="url"><img src="TeachingTipHeroContentSample.jpg" height="600"></a>
+
+### Add an icon
+
+An icon can be added to the side of the title and subtitle using the IconSource property.
```

Comment:
So how much control does an app developer have over the size of the icon? How is the default size determined? What's the default margin between icon and title/subtitle, and can app developers customize it?

Really, I'm questioning the need for this property because it feels like a can of worms. Is a customer clamoring for it?

Reply:
Great question, @YuliKl. The icon property will take whatever size the icon is and will move the title and subtitle over to accommodate. Note that the icon will start to become cropped when it begins hitting the tip's max size. 16px, 24px, and 32px are our design recommended icon sizes. This code preview is now outdated as I have now added this to the guidance. 

![image](https://user-images.githubusercontent.com/16964652/53135493-3b3bfc00-3530-11e9-91e0-391832ee0128.png)

The margin between an icon and the title/subtitle cannot be manipulated directly and there have been no requests yet for this knob to be included. However, by the logic noted above, there is already a possible workaround where the icon image itself can be created to include extra padding within the icon since there is no strict size enforced.